Formula/p/python-matplotlib.rb

```diff
@@ -95,25 +76,38 @@ class PythonMatplotlib < Formula
   end
 
   def python3
-    which("python3.12")
+    "python3.12"
   end
 
   def install
-    resource("contourpy").stage do
-      system python3, "-m", "pip", "install", *std_pip_args, "."
-    end
-
+    # `matplotlib` needs extra inputs to use system libraries.
+    # Ref: https://github.com/matplotlib/matplotlib/blob/v3.8.3/doc/users/installing/dependencies.rst#use-system-libraries
+    # TODO: Update build to use `--config-settings=setup-args=...` when `matplotlib` switches to `meson-python`.
+    ENV["MPLSETUPCFG"] = buildpath/"mplsetup.cfg"
     (buildpath/"mplsetup.cfg").write <<~EOS
       [libs]
       system_freetype = true
       system_qhull = true
     EOS
 
-    system python3, "-m", "pip", "install", *std_pip_args, "."
+    virtualenv_install_with_resources
+  end
+
+  def caveats
+    <<~EOS
+      `matplotlib` is no longer linked into shared site-packages to avoid contamination
+      and align with Homebrew's documented requirements for Python library formulae.
+      To use the library, you will need to update your PYTHONPATH to include the directory:
+      #{opt_libexec/Language::Python.site_packages(python3)}
+
+      See https://docs.brew.sh/Python-for-Formula-Authors
+      See https://docs.brew.sh/Homebrew-and-Python#pep-668-python312-and-virtual-environments
+      See https://github.com/Homebrew/homebrew-core/issues/157500
+    EOS
   end
```

Comment:
Enough installs that I am starting with caveat and maybe add deprecation phase afterward:
```
install-on-request: 1,224 (30 days), 2,049 (90 days), 2,495 (365 days)
```

This is only a library with no `bin` scripts.